src/lib/decodeData.ts

```diff
@@ -109,16 +111,25 @@ export const isValidTuple = (valueType: string, valueContent: string) => {
       );
     }
 
-    if (
-      valueContentParts[i].slice(0, 2) === '0x' &&
-      !isHex(valueContentParts[i])
-    ) {
-      throw new Error(
-        `Invalid tuple for valueType: ${valueType} / valueContent: ${valueContent}. valueContent of type: ${valueContentParts[i]} is not a valid hex value`,
-      );
-    }
+    const isValueContentHexLiteral = valueContentParts[i].slice(0, 2) === '0x';
+
+    if (isValueContentHexLiteral) {
+      if (!isHex(valueContentParts[i])) {
```

Comment:
```suggestion
    if (isValueContentHexLiteral && !isHex(valueContentParts[i])) {
```

Reply:
I think it can be simplified:

```
const isValueContentHexLiteral = valueContentParts[i].slice(0, 2) === '0x';
if (isValueContentHexLiteral) {
      if (!isHex(valueContentParts[i])) {

// ...
```

I think you can do all of this in one line, you can use https://web3js.readthedocs.io/en/v1.2.11/web3-utils.html#ishexstrict -> this will check for 0x AND if it is hexadecimal.